docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,261 @@ Documentation for other releases can be found at
 
 # kubectl overview
 
+Use this overview of the `kubectl` command line interface to help you start running commands against Kubernetes clusters. We will quickly cover `kubectl` syntax, describe the command operations, and provide common examples. For details about each command, including all the supported options, see the [kubectl](kubectl/kubectl.md) reference documentation.
+
 **Table of Contents**
 
-- [kubectl overview](#kubectl-overview)
-  - [Overview](#overview)
-  - [Common Operations](#common-operations)
-  - [Kubectl Operations](#kubectl-operations)
-  - [Resource Types](#resource-types)
+ * [Syntax](#syntax)
+ * [Operations](#operations)
+ * [Resource types](#resource-types)
+ * [Output options](#output-options)
+ * [Examples: Common operations](#examples-common-operations)
+
+TODO: Auto-generate this file to ensure it's always in sync with any `kubectl` changes, see [#14177](https://github.com/kubernetes/kubernetes/pull/14177).
+
+## Syntax
+
+Use the following syntax to run `kubectl` commands from your terminal window:
+
+```
+kubectl [command] [TYPE] [NAME] [options]
```

Comment:
I see. I was trying to be consistent with the other docs and got `[options]` from all the auto generated .md files ([kubectl get](https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/kubectl/kubectl_get.md)) because they all refer to the flags as "options". 

I agree, ill change it back to `[flags]`. We should align with kubectl-conventions.md and kubectl --help output and also get all the auto generated files updated so that all our content consistently uses `[flags]` (in otherwords, in all these files, we should replace "options" with "flags" [kubectl](https://github.com/kubernetes/kubernetes/tree/master/docs/user-guide/kubectl)).